pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,163 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk
-, wrapGAppsHook3
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  autoPatchelfHook,
+  wrapGAppsHook3,
+  glib-networking,
+  gst_all_1,
+  libappindicator,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm,
+  webkitgtk,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
   pname = "dorion";
   version = "5.0.1";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  webkitgtk' = webkitgtk.override { enableExperimental = true; };
+
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${version}";
+    hash = "sha256-x0WLDeR0By2hjoMDZvGXVR83+xICHU8YEd5pWf5nkvs=";
+
+    nativeBuildInputs = [ pnpm.configHook ];
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/1bbbab09ccb98574292ba5b658732e13db7b31bd/shelter.js";
+    hash = "sha256-MHb4IwmLmcwjE9LUDsUD3cU1B0LxGTDzpc0lvnWmMfY=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
```

Comment:
#335487

Reply:
Should I uh remove the other `binaryByteCode` or just keep it in as well? (Also, [shelter](https://github.com/uwu/shelter) has two repos the one I grabbed from is [shelter-builds](https://github.com/uwu/shelter-builds), equivalent to what [upstream does](https://github.com/SpikeHD/Dorion/blob/7c55c5cf32c1c9f8235a63c2c60b0ae764b07843/package.json#L20) Shelter-builds is just builds that the devs build and redistribute themselves.

Reply:
Would this evaluate if I changed it to what your issue suggests? :P
Also, should I remove the other 'binaryByteCode' or just keep it? Also, [shelter](https://github.com/uwu/shelter) has two repos the one I grabbed from is [shelter-builds](https://github.com/uwu/shelter-builds), equivalent to what [upstream does](https://github.com/SpikeHD/Dorion/blob/7c55c5cf32c1c9f8235a63c2c60b0ae764b07843/package.json#L20) Shelter-builds is just builds that the devs build and redistribute themselves.

Reply:
> Would this evaluate if I changed it to what your issue suggests? 

Not yet :)
I am marking here because your issue is the first one affected by the suggestion I issued.

> Also, should I remove the other 'binaryByteCode' or just keep it? 

You can do this as you wish.
My suggestion would be add a commentary: 
`. . . binaryBytecode # actually, minified JS`

> Also, [shelter](https://github.com/uwu/shelter) has two repos the one I grabbed from

Being pedantic, the best course of action would be grab the real human-made source code and compile ourselves (or whatever is better - I guess the minified code is equivalent to original one).

But let's do this another day...

Reply:
Did the comment :P